controllers/componentdetectionquery_controller.go

```diff
@@ -176,13 +173,15 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 					metrics.HandleRateLimitMetrics(err, metricsLabel)
 					log.Error(err, fmt.Sprintf("Unable to get main branch of Github Repo %v ... %v", source.URL, req.NamespacedName))
 					retErr := fmt.Errorf("unable to get default branch of Github Repo %v, try to fall back to main branch, failed to get main branch... %v", source.URL, req.NamespacedName)
-					r.SetCompleteConditionAndUpdateCR(ctx, req, &componentDetectionQuery, copiedCDQ, retErr)
+					r.SetCompleteConditionAndUpdateCR(ctx, req, &componentDetectionQuery, retErr)
 					return ctrl.Result{}, nil
 				} else {
 					source.Revision = "main"
 				}
 			}
 		}
+		// set in the CDQ spec
+		componentDetectionQuery.Spec.GitSource.Revision = source.Revision
```

Comment:
We shouldn’t be modifying things in the CR spec, anything we return should be in the status only.